CONTRIBUTING.md

```diff
@@ -10,55 +10,146 @@ For general contribution and community guidelines, please see the [community rep
   - [ ] Bump version in [`CHANGELOG.md`](CHANGELOG.md).
+  
+- [ ] [`Build`](./build) the release. Output is placed in the `dist` folder.
+   - [ ] Run smoke tests using the binaries in their respective output directories on
+         their appropriate platform.
+        - Linux:   `dist/summon-linux_linux_amd64`
+        - MacOS:   `dist/summon_darwin_amd64`
+        - Windows: `dist/summon_windows_amd64`
+
 - [ ] Create a draft release:
   - [ ] Create an [annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_annotated_tags)
-of the merged commit with the name `v<version>` and push it to the repository.
+        of the merged commit with the name `v<version>` and push it to the repository.
+        For example: `git tag -a v<version> -m "Release of v<version>"  `
   - [ ] Using the GitHub UI, create a release using the tag created earlier as the starting point.
   - [ ] Name the release the same as the tag.
   - [ ] Include in the release notes all changes from CHANGELOG that are being released.
-- [ ] Attach the relevant assets to the release:
-  - [ ] [`Build`](./build) the release.
-  - [ ] Attach `dist/summon-darwin-amd64.tar.gz` to release.
-  - [ ] Attach `dist/summon-linux-amd64.tar.gz` to release.
-  - [ ] Attach `dist/summon-windows-amd64.tar.gz` to release.
-  - [ ] Attach `dist/summon_v*.rpm` to release.
-  - [ ] Attach `dist/summon_v*.deb` to release.
-  - [ ] Attach `dist/CHANGELOG.md` to release.
-  - [ ] Edit `dist/SHA256SUMS.txt` to include only the attached files above.
-  - [ ] Attach `dist/SHA256SUMS.txt` to the release.
-- [ ] Publish the release as a "pre-released".
-- [ ] TBD: Perform smoke tests of all attached files in release.
-- [ ] Publish the release as a regular release.
+    - [ ] Attach the relevant assets to the release, generated previously by `./build`:
+      - [ ] Attach `dist/summon-darwin-amd64.tar.gz` to release.
+      - [ ] Attach `dist/summon-linux-amd64.tar.gz` to release.
+      - [ ] Attach `dist/summon-windows-amd64.tar.gz` to release.
+      - [ ] Attach `dist/summon_v*.rpm` to release.
+      - [ ] Attach `dist/summon_v*.deb` to release.
+      - [ ] Attach `dist/CHANGELOG.md` to release.
+      - [ ] Attach `dist/SHA256SUMS.txt` to the release.
+  - [ ] Publish the release as a "pre-released".
+  
 - [ ] Update homebrew tools
   - [ ] In [`cyberark/homebrew-tools`](https://github.com/cyberark/homebrew-tools) repo, update
-  the [`summon.rb` formula](https://github.com/cyberark/homebrew-tools/blob/master/summon.rb#L4-L6) with a PR
-  using the file `dist/summon.rb`.
-  - [ ] Once the PR is merged, verify that summon works by smoke testing it on OSX.
+        the [`summon.rb` formula](https://github.com/cyberark/homebrew-tools/blob/master/summon.rb#L4-L6) with a PR
+        using the file `dist/summon.rb`.
+        
+- [ ] Publish the release as a regular release.
+    - [ ] Perform smoke tests, using this release, on the following platforms:
```

Comment:
Smoke tests should happen before switching from pre-release to release

Reply:
Previously, we had instructions for running tests after the pre-release, and after the release.
Currently, I changed the instructions to run smoke tests before the pre release and after the release. 
Can you clarify which is the correct order?

Reply:
Clarified instructions